pkg/nvml/lib.go

```diff
@@ -43,27 +53,45 @@ type library struct {
 	dl       dynamicLibrary
 }
 
+var _ Interface = (*library)(nil)
+
 // libnvml is a global instance of the nvml library.
-var libnvml = library{
-	path: defaultNvmlLibraryName,
-	dl:   dl.New(defaultNvmlLibraryName, defaultNvmlLibraryLoadFlags),
-}
+var libnvml = newLibrary()
 
-var _ Interface = (*library)(nil)
+func New(opts ...LibraryOption) Interface {
+	return newLibrary(opts...)
+}
 
-// GetLibrary returns a the library as a Library interface.
-func (l *library) GetLibrary() Library {
+func newLibrary(opts ...LibraryOption) *library {
+	l := &library{}
+	l.init(opts...)
 	return l
 }
 
-// GetLibrary returns a representation of the underlying library that implements the Library interface.
-func GetLibrary() Library {
-	return libnvml.GetLibrary()
+func (l *library) init(opts ...LibraryOption) {
+	o := libraryOptions{}
+	for _, opt := range opts {
+		opt(&o)
+	}
+
+	if o.path == "" {
+		o.path = defaultNvmlLibraryName
+	}
+	if o.flags == 0 {
+		o.flags = defaultNvmlLibraryLoadFlags
+	}
+
+	l.path = o.path
+	l.dl = dl.New(o.path, o.flags)
+}
+
+func (l *library) GetExtendedInterface() ExtendedInterface {
```

Comment:
I'm not sold on `GetExtendedInterface` and would prefer `GetExtensions()`, but this is not a blocker. I'm happy to keep it as is and revisit this later once we have some usage examples.

Reply:
Updated as suggested.

Reply:
Actually, I went for a hybrid -- the interface is called `ExtendedInterface` to be symmetrical with `Interface`, but the method call is `Extensions()` (which is the only actual symbol anyone will see in practice), e.g.:
```
nvml.Extensions().LookupSymbol("symbol")
```
or
```
envml := nvml.Extensions()
envml.LookupSymbol("symbol")
```

Its possible that the type itself might be embedded in a struct, but I only see that happening in testing (if at all) and I like the symmetry of that with `Interface` for the core API.

Reply:
That sounds fine. I'm not too concerned about the type name, since as you mention that isn't visible at the call site.